identity/identity.go

```diff
@@ -75,20 +75,16 @@ type XRHID struct {
 // Key the key for the XRHID in that gets added to the context
 const Key identityKey = iota
 
-func getErrorText(code int, reason string) string {
-	return http.StatusText(code) + ": " + reason
-}
-
-func doError(w http.ResponseWriter, code int, reason string) error {
-	http.Error(w, getErrorText(code, reason), code)
-	return errors.New(reason)
-}
-
 // Get returns the identity struct from the context
 func Get(ctx context.Context) XRHID {
 	return ctx.Value(Key).(XRHID)
```

Comment:
This panics the program if the typecast fails. Could you include a simple check on the cast and return a zero-value XHRID struct instead?

```suggestion
        x, ok := ctx.Value(Key).(XRHID)
        if !ok {
            return XRHID{}
        }
	return x
```

Reply:
Absolutely.

Reply:
Although I have one idea - the reason why you can ever see panic is because `Key` is an exported value. It should not have been ever exported, this allows callers to mess around with the value. I am going to change my patch to unexport this key for v2, meaning there is no way to be able to save a value into the context under this key for anyone except this package. Therefore panic will never happen.

The reason for others to use the Key directly was I think because the API was not designed cleanly and there was no good way of implementing your own middleware easily. This will be actually now possible using many new exported functions which will allow others to implement middleware easily.

What you think? Is my understanding of this problem correct?